src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,2068 @@
+            int y = 2;
+            int[] a = { 1,2 }
+            " + code + @"
+
+            int z = 4;
+        }
+    }
+
+    static class ClassC
+    {
+        internal static int MethodM(int a, int b)
+            => a * b;
+    }
+}";
+        }
+
+        #region ArgumentListOfMethodInvocation
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation1()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM($$x, y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation2()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x$$, y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation3()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x,$$ y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation4()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, $$y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation5()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y$$)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation6()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y)$$");
+
+            var expected = code.Replace("$$", ";$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation_MissingParen()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y$$");
+
+            var expected = code.Replace("$$", ";$$");
+
+            VerifyTypingSemicolon(code, expected);
```

Comment:
1. this case surprises me.  i thought part of the intent here was that we would be 'completing the missing tokens' here.  So i would have expected the missing parent to be inserted.
2. using code.replace here just makes things confusing to me as i now need to figure out in my head what the actual final form is.  Why not follow the pattern of everything above and just specify the 'expected' code fully?

Reply:
1. Insertion of missing delimiters has a bunch of nasty edge cases. I plan to file a follow-up issue to start implementing subsets of this functionality, but I didn't want it to block a much easier primary case where automatic brace completion is used.
2. I wanted to extract a helper equivalent to `VerifyTypingSemicolon(code)` where `expected` is calculated inside the method, but I couldn't think of a good name.

Reply:
I'm considering this resolved with the new `VerifyNoSpecialSemicolonHandling` helper.